packages/flutter/lib/src/material/toggle_buttons.dart

```diff
@@ -1299,8 +1309,25 @@ class _SelectToggleButtonRenderObject extends RenderShiftedBox {
       rrect.brRadiusY * 2,
     );
 
+    final Paint leadingPaint = leadingBorderSide.toPaint();
+    // Only one button.
+    if (isFirstButton && isLastButton) {
+      final Path leadingPath = Path();
+      final double startX = (rrect.brRadiusX == 0.0) ? outer.right : rrect.right - rrect.brRadiusX;
```

Comment:
If the radius is zero, we should paint a little more to connect the start and stop ends.
Otherwise, there will be a gap like below:
![Uploading image.png…]()

Reply:
Nice, good catch here